pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/AttributeAxisIterator.java

```diff
@@ -24,7 +26,9 @@
      */
     public AttributeAxisIterator(ElementNode startNodeInfo) {
         this.startNodeInfo = startNodeInfo;
-        this.iterator = new net.sourceforge.pmd.lang.ast.xpath.AttributeAxisIterator(startNodeInfo.node);
+        this.iterator = startNodeInfo.node instanceof net.sourceforge.pmd.lang.ast.xpath.AttributeNode
+                ? ((net.sourceforge.pmd.lang.ast.xpath.AttributeNode) startNodeInfo.node).getAttributeIterator()
+                : new net.sourceforge.pmd.lang.ast.xpath.AttributeAxisIterator(startNodeInfo.node);
```

Comment:
this exact logic is repeated the change you made to `NodeInfoPanelController` and already present in `DocumentNavigator.getAttributeAxisIterator`... it seems like we should refactor this away

Reply:
Note that in the case of `getXPathNodeName()` we added it directly to the Node interface and provided a default implementation in AbstractNode, which was deemed compatible enough at the time. Doing the same thing here would make sense I think

Reply:
I support doing the same here, yes

Reply:
Done!